test/converters/xcresult/xcresult.go

```diff
@@ -0,0 +1,188 @@
+		testID := strings.Split(test.TestIdentifier, "/")[0]
+		tests[testID] = append(tests[testID], test)
+	}
+	return tests
+}
+
+type TestableSummary struct {
+	TargetName      string
+	TestKind        string
+	TestName        string
+	TestObjectClass string
+	Tests           []Test
+}
+
+type FailureSummary struct {
+	FileName           string
+	LineNumber         int
+	Message            string
+	PerformanceFailure bool
+}
+
+type Subtest struct {
+	Duration         float64
+	TestStatus       string
+	TestIdentifier   string
+	TestName         string
+	TestObjectClass  string
+	Subtests         Subtests
+	FailureSummaries []FailureSummary
+}
+
+func (st Subtest) Failure() (message string) {
+	prefix := ""
+	for _, failure := range st.FailureSummaries {
+		message += fmt.Sprintf("%s%s:%d - %s", prefix, failure.FileName, failure.LineNumber, failure.Message)
+		prefix = "\n"
+	}
+	return
+}
+
+type Subtests []Subtest
+
+func (sts Subtests) FailuresCount() (count int) {
+	for _, test := range sts {
+		if len(test.FailureSummaries) > 0 {
+			count++
+		}
+	}
+	return count
+}
+
+func (sts Subtests) TotalTime() (time float64) {
+	for _, test := range sts {
+		time += test.Duration
+	}
+	return time
+}
+
+type Test struct {
+	Subtests Subtests
+}
+
+//
+////
+
+// Junit4 XML
+
+type Junit4XML struct {
+	XMLName    xml.Name `xml:"testsuites"`
+	TestSuites []TestSuite
+}
+
+type TestSuite struct {
+	XMLName   xml.Name `xml:"testsuite"`
+	Name      string   `xml:"name,attr"`
+	Tests     int      `xml:"tests,attr"`
+	Failures  int      `xml:"failures,attr"`
+	Errors    int      `xml:"errors,attr"`
+	Time      float64  `xml:"time,attr"`
+	TestCases []TestCase
+}
+
+type TestCase struct {
+	XMLName   xml.Name `xml:"testcase"`
+	Name      string   `xml:"name,attr"`
+	ClassName string   `xml:"classname,attr"`
+	Time      float64  `xml:"time,attr"`
+	Failure   string   `xml:"failure,omitempty"`
+}
+
+//
+////
+
+type Handler struct {
+	files                  []string
+	testSummariesPlistPath string
+}
+
+func (h *Handler) SetFiles(files []string) {
+	h.files = files
+}
+func (h *Handler) Detect() bool {
+	for _, file := range h.files {
+		if strings.HasSuffix(file, ".xcresult") {
+			h.testSummariesPlistPath = filepath.Join(file, "TestSummaries.plist")
+			return true
+		}
+	}
+	return false
+}
+
+func (h *Handler) XML() ([]byte, error) {
+	data, err := fileutil.ReadBytesFromFile(h.testSummariesPlistPath)
+	if err != nil {
+		return nil, err
+	}
+
+	var plistData TestSummaryPlist
+	if _, err := plist.Unmarshal(data, &plistData); err != nil {
+		return nil, err
+	}
+
+	var xmlData Junit4XML
+	for testID, tests := range plistData.Tests() {
+		testSuite := TestSuite{
+			Name:     testID,
+			Tests:    len(tests),
+			Failures: tests.FailuresCount(),
+			Time:     tests.TotalTime(),
+		}
+
+		for _, test := range tests {
+			testSuite.TestCases = append(testSuite.TestCases, TestCase{
+				Name:      test.TestName,
+				ClassName: testID,
```

Comment:
Why is the ClassName and the TestSuite name testID too?

Reply:
In the junit xml it is grouped like that, I think testsuites are grouped in the named teststuites branch